internal/common/monitors_runner.go

```diff
@@ -0,0 +1,76 @@
+package common
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/openshift/assisted-service/pkg/leader"
+
+	"github.com/sirupsen/logrus"
+)
+
+type Monitor struct {
+	Name     string
+	Interval time.Duration
+	Exec     func()
+}
+
+type MonitorRunnerWithLeader struct {
+	log      logrus.FieldLogger
+	done     chan struct{}
+	monitors []Monitor
+	elector  leader.ElectorInterface
+	ctx      context.Context
+}
+
+func NewMonitorRunnerWithLeader(log logrus.FieldLogger, monitors []Monitor, elector leader.ElectorInterface) *MonitorRunnerWithLeader {
```

Comment:
how is it different from the thread package that we have?

Reply:
first of all it takes list of monitors.
and spawns routine per each monitor without depending on others.
Uses leader to start running.
I started with thread package but it was not very helpful, it blocked me. I think this one is more helpful in my case

Reply:
For me it looks like a different run function is the only difference.

Reply:
It has common things i agree but still the idea is different
Thread manages one goroutine at a time and i need code that will manage list of routines.
What is happening is that when leader is taken runs callback so to be able to have one leader for all monitors i need to provide one functions that manages them. And need a way to stop all of them. 
This code provides me this capability and thread doesn't.





I can add this code to thread

Reply:
I guess that this is another implementation of our commont thread interface, right?
It does not really start a thread, but calls the leaderElection logic which starts the actual thread.
Am I right @tsorya ?

Reply:
almost. As i wrote there are 2 main differences.
1. Leader
2. It takes list and manages list of go routines